### PR TITLE
Add ranked and sorting options to RARBG settings

### PR DESCRIPTION
**Pull request by fernandog**
Ranked: If checked will only return rartv, ettv, and scene releases released internally on rarbg. Otherwise returns all results from rarbg site.
Sorting: By seeders, leechers, or newest first

### CODE REVIEW

**Comment by fernandog** [comment]
will squash this later

**Comment by miigotu** [comment]
@fernandog I got your texts, but I haven't checked the diff yet. Maybe we should add sorting to the options too? It was by newest (last) before, I changed it to by seeders like the other providers, but there is also by leechers.

File: gui/slick/interfaces/default/config_providers.tmpl [hunk 1]
```diff
@@ -459,6 +459,18 @@ var show_nzb_providers = #if $sickbeard.USE_NZBS then "true" else "false"#;
                         </div>
                         #end if
 
+                        #if $hasattr($curTorrentProvider, 'ranked'):
+                        <div class="field-pair">
+                            <label for="${curTorrentProvider.getID()}_ranked">
+                                <span class="component-title">Ranked download</span>
+                                <span class="component-desc">
+                                    <input type="checkbox" name="${curTorrentProvider.getID()}_ranked" id="${curTorrentProvider.getID()}_ranked" #if $curTorrentProvider.ranked then "checked=\"checked\"" else ""#/>
+                                    <p>only download ranked torrents (internal releases)</p>
```
Comment (miigotu) [inline gui/slick/interfaces/default/config_providers.tmpl]: Capital O for only?

Comment (fernandog) [inline gui/slick/interfaces/default/config_providers.tmpl]: all the others are lower

**Comment by miigotu** [comment]
All looks good, just need the capital O, =P.

File: sickbeard/webserve.py [hunk 2]
```diff
@@ -4478,6 +4478,19 @@ def saveProviders(self, newznab_string='', torrentrss_string='', provider_order=
...
+            if hasattr(curTorrentProvider, 'ranked'):
+                try:
+                    curTorrentProvider.ranked = config.checkbox_to_value(
+                        kwargs[curTorrentProvider.getID() + '_ranked'])
+                except:
+                    curTorrentProvider.ranked = 0
+
+            if hasattr(curTorrentProvider, 'sorting'):
+                try:
+                    curTorrentProvider.sorting = kargs[curTorrentProvider.getID() + '_sorting']
```
Comment (fernandog) [inline sickbeard/webserve.py]: @miigotu I don't know for sure if this is correct. Can you confirm please? Also please check the latests commits about "sorting"

**Comment by fernandog** [comment]
@miigotu getting this error. don't know how to fix it:

```
ParseError: 

Long-form placeholders - ${}, $(), $[], etc. are not valid inside expressions. Use them in top-level $placeholders only.
Line 454, column 40 in file /home/osmc/SickRage/gui/slick/interfaces/default/config_providers.tmpl
```

File: gui/slick/interfaces/default/config_providers.tmpl [hunk 3]
```diff
@@ -420,6 +420,45 @@ var show_nzb_providers = #if $sickbeard.USE_NZBS then "true" else "false"#;
...
+                                </span>
+                            </label>
+                        </div>
+                        #end if
+
```
Comment (miigotu) [inline gui/slick/interfaces/default/config_providers.tmpl]: #if $hasattr($curTorrentProvider, 'sorting'):

File: gui/slick/interfaces/default/config_providers.tmpl [hunk 4]
```diff
@@ -420,6 +420,45 @@ var show_nzb_providers = #if $sickbeard.USE_NZBS then "true" else "false"#;
...
+#set $sorting_text = {'last': "Last", 'seeders': "Seeders", 'leechers': "Leechers"}
+#for $curAction in ('last', 'seeders', 'leechers'):
+    #set $selected = $html_selected if ${curTorrentProvider.getID()}_sorting == $curAction else ''
+                                    <option value="$curAction"$selected>$sorting_text[$curAction]</option>
+#end for
+                                    </select>
+                                </span>
+                            </label>
+                        </div>
+
```
Comment (miigotu) [inline gui/slick/interfaces/default/config_providers.tmpl]: #end if

**Comment by miigotu** [comment]
@fernandog those should fix your parse error.

File: gui/slick/interfaces/default/config_providers.tmpl (hunk 3 above)
Comment (fernandog) [inline gui/slick/interfaces/default/config_providers.tmpl]: just added. it's working but it's not saving...

**Comment by miigotu** [comment]
Had to delete the comments to be able to read the diff easier, i'll try and find why it isnt saving.

**Comment by fernandog** [comment]
fixed! typo in kwags > kwargs

File: sickbeard/webserve.py (hunk 2 above)
Comment (miigotu) [inline sickbeard/webserve.py]: curTorrentProvider.sorting = str(kwargs[curTorrentProvider.getID() + '_sorting']).strip()

**Comment by miigotu** [comment]
Did you also wrap it in str().strip() ?

curTorrentProvider.sorting = str(kwargs[curTorrentProvider.getID() + '_sorting']).strip()

**Comment by fernandog** [comment]
@miigotu dont need because its a select field not input text

**Comment by miigotu** [comment]
Still makes it safer, better to be strict in case someone edits the config.ini directly.

**Pull request merged**